app/src/main/java/com/github/se/studybuddies/ui/groups/MembersList.kt

```diff
@@ -0,0 +1,56 @@
+package com.github.se.studybuddies.ui.groups
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.github.se.studybuddies.R
+import com.github.se.studybuddies.database.DbRepository
+import com.github.se.studybuddies.navigation.NavigationActions
+import com.github.se.studybuddies.navigation.Route
+import com.github.se.studybuddies.ui.shared_elements.GoBackRouteButton
+import com.github.se.studybuddies.ui.shared_elements.Sub_title
+import com.github.se.studybuddies.ui.shared_elements.TopNavigationBar
+import com.github.se.studybuddies.ui.theme.White
+import com.github.se.studybuddies.viewModels.GroupViewModel
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun MembersList(
+    groupUID: String,
+    groupViewModel: GroupViewModel,
+    navigationActions: NavigationActions,
+    db: DbRepository
+) {
+
+  if (groupUID.isEmpty()) return
+  val isBoxVisible = remember { mutableStateOf(true) }
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().background(White),
+      topBar = {
+        TopNavigationBar(
+            title = { Sub_title(stringResource(R.string.add_member)) },
+            navigationIcon = {
+              GoBackRouteButton(navigationActions = navigationActions, Route.GROUPSHOME)
```

Comment:
I realized that on the 5 different screens of group settings (modify group, ..., delete group) when pressing the back button we navigate to GroupsHome, which feels a bit weird. Instead of this line, can you just call `navigationActions.goBack` (for the 5 screens)? This way the app will just navigate to the previous screen and it will feel much better.

Reply:
Yes you are correct, I change the function used for the group so that it goes back to last screen.